react/RedirectList.tsx

```diff
@@ -240,16 +241,23 @@ const RedirectList: React.FC<Props> = ({
               })
             }
 
-            const innerNext = innerData?.redirect?.listRedirects.next
-            const next = innerNext ?? data?.redirect?.listRedirects.next
-
-            if (!filteredRedirects.length && next) {
-              const { data } = await refetch({ limit: REDIRECTS_LIMIT, next })
-              handleInputSearchChange(e, data)
-            } else {
-              setRedirectList(filteredRedirects)
+            let result: Redirect[] = []
+            if (term.length) {
+              const { data } = await client.query({
+                query: RedirectWithoutBinding,
+                variables: {
+                  path: term,
+                },
+              })
+              if (data?.redirect?.get) {
+                result = [data.redirect.get]
+              }
             }
 
+            const combinedResults = [...filteredRedirects, ...result]
+
+            setRedirectList(combinedResults)
+
```

Comment:
Tentando separar a chamada numa função e adicionando um early return 

```suggestion

const searchExactMatchRedirect = async (term: string) => {
  if (!term) return [];

  const { data } = await client.query({
    query: RedirectWithoutBinding,
    variables: {
      path: term,
    },
  });
  
  const { get: redirectFound } = data?.redirect ?? {};

  return redirectFound ? [redirectFound] : [];
};

const exactMatchRedirect = await searchExactMatchRedirect(term);
exactMatchRedirect.length > 0 ?? setRedirectList([...filteredRedirects, ...exactMatchRedirect]);

```

Reply:
Nice! Suggestion applied.